From 29bd3960ebb9b37febdd3455259a8aaa674a9409 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 110/296] When monkey test, the object of mHost throws
 NullPointerException.

Fail log: java.lang.NullPointerException: Attempt to invoke virtual
method 'android.app.LoaderManagerImpl
android.app.FragmentHostCallback.getLoaderManager(java.lang.String, boolean, boolean)' on a null object reference
at android.app.Fragment.onDestroy(Fragment.java:1603)  	at android.app.Fragment.performDestroy(Fragment.java:2433)
at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1098)
at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1148)
at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1130)
at android.app.FragmentManagerImpl.dispatchDestroy

Root cause:
The object of mHost is null, throws NullPointerException.

Solution:
Add the condition of whether mHost is null
---
 core/java/android/app/Fragment.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/java/android/app/Fragment.java b/core/java/android/app/Fragment.java
index 93773454424..18025235a16 100644
--- a/core/java/android/app/Fragment.java
+++ b/core/java/android/app/Fragment.java
@@ -1781,7 +1781,9 @@ public class Fragment implements ComponentCallbacks2, OnCreateContextMenuListene
         //        + " mLoaderManager=" + mLoaderManager);
         if (!mCheckedForLoaderManager) {
             mCheckedForLoaderManager = true;
-            mLoaderManager = mHost.getLoaderManager(mWho, mLoadersStarted, false);
+            if (mHost != null) {
+                mLoaderManager = mHost.getLoaderManager(mWho, mLoadersStarted, false);
+            }
         }
         if (mLoaderManager != null) {
             mLoaderManager.doDestroy();
-- 
2.11.0

